### PR TITLE
Adds solar and wind generators to the model

### LINKED ISSUE #5: Add renewable resources to the model
This issue may be closed when there is a pull request that adds the following generators

 - [ ] solar (i.e., utility scale)
 - [ ] onshore wind

AND resource availability data 
 
- [ ] wind speed (PJM)
- [ ] wind speed (MISO)
- [ ] solar irradiation (PJM)
- [ ] solar irradiation (MISO)

For the purposes of resolving this issue, these data may be taken from the geographic center of these two zones. This will likely change in the future.

### PR DESCRIPTION
**Pull request by samgdotson**
This pull request closes #5 by adding solar and wind generators to the illinois model notebook. Included are the following
- A notebook that extracts data from NREL for solar and wind resources
- A section in the `illinois_model.ipynb` notebook that explains the data manipulations (specifically, we just want the _shape_ of the generation so it can be scaled by the optimal capacity, later).
- The wind and solar generators are added to the model.
- The model runs successfully.

Currently, the solar and wind resources are only considered at a single point in each modeled region (the centroid). In the future, we may consider expanding the distribution of these resources by increasing the number of buses, for example (each bus does not have to have a load). 

Edit:
This PR also directly adds the relevant datasets.

### CODE REVIEW

File: notebooks/nrel_data_download.ipynb
Comment (lshaver) [inline notebooks/nrel_data_download.ipynb:1]: Instructions for adding the NREL api key to the conda environment would be helpful, including a way to change the personal data dictionary as needed in the nrel_data_api.py function.

**Review by lshaver — APPROVED** [review]
There are some minor typos/errors that need to be corrected in future pull requests but otherwise everything seems to be working well.

**Comment by lshaver** [comment]
In reviewing the code again I see this assumption which I'm a bit concerned about: "However, if we assume that temperature has no effect on panel performance, then the output power and incident power are linearly related." With rising temperatures and more solar on the grid, ignoring the reduction in solar output on hot summer days will distort the model. Should we address this?

**Comment by samgdotson** [comment]
Average temperatures are rising, but for a first order model (which is what we're trying to achieve right now), it doesn't make sense to incorporate. Even if we did, the effect would be quite small. We can circle back on this in the future though, if you think it's important.

**Comment by lshaver** [comment]
Looks like there's about a [5% loss in performance](https://www.cleanenergyreviews.info/blog/most-efficient-solar-panels) going from panel rating temperature to 35 degC, the average high in Chicago (in July). During peak hours this could be significant, but I agree it's probably not worth including in the model at this point. It could be an interesting sensitivity to do later, possibly even incorporating future and extreme weather data.

**Pull request merged**